pyuvdata/parameter.py

```diff
@@ -930,43 +939,84 @@ def __init__(
         value=None,
         spoof_val=None,
         description="",
-        frame="itrs",
-        ellipsoid=None,
-        acceptable_range=None,
         tols=1e-3,
     ):
         super(LocationParameter, self).__init__(
             name,
             required=required,
             value=value,
             spoof_val=spoof_val,
-            form=3,
             description=description,
-            expected_type=float,
-            acceptable_range=acceptable_range,
+            expected_type=tuple(allowed_location_types),
+            strict_type_check=True,
+            acceptable_range=None,
             tols=tols,
         )
-        self.frame = frame
 
-        if frame == "mcmf" and ellipsoid is None:
-            ellipsoid = "SPHERE"
+    @property
+    def frame(self):
+        """Get the frame."""
+        if isinstance(self.value, EarthLocation):
+            return "itrs"
+        elif hasmoon and isinstance(self.value, MoonLocation):
+            return "mcmf"
+
+    @property
+    def ellipsoid(self):
+        """Get the ellipsoid."""
+        if isinstance(self.value, EarthLocation):
+            return None
+        elif hasmoon and isinstance(self.value, MoonLocation):
+            return self.value.ellipsoid
 
-        self.ellipsoid = ellipsoid
+    def xyz(self):
+        """Get the body centric coordinates in meters."""
+        if self.value is None:
+            return None
+        else:
+            if hasmoon and isinstance(self.value, MoonLocation):
+                centric_coords = self.value.selenocentric
+            else:
+                centric_coords = self.value.geocentric
+            return units.Quantity(centric_coords).to("m").value
```

Comment:
Note that this is not currently turned into a property like `location_xyz` on the Telescope object, unlike e.g. the `lat_lon_alt` method which is turned into the `location_lat_lon_alt` property on the Telescope object. Initially this was because I was trying to push users to just use the EarthLocation object, but there are several places in the code where I call this method because getting the xyz as a simple numpy array (rather than Quantities) is useful. Thoughts? @steven-murray @kartographer

Reply:
I would refrain from as much bespoke functionality as possible, and go with your inclination towards pushing users to use the `EarthLocation` object. "There should be one-- and preferably only one --obvious way to do it."

Reply:
👍 for having this as a convenience function.

Reply:
The annoying thing is how hard it is to get the xyz location from an astropy EarthLocation object. I'd be happy if it'd just give it to you as a 3-element quantity array, but it doesn't. You have to ask for x, y, and z separately. Or you can do el_obj.geocentric, which gives you a list of 3 Quantities, which still takes work to get into a useful shape.

Reply:
So, as convenient as it might be to have `.xyz` as a property, I think it's less explicit than using the `EarthLocation` -- the name doesn't tell you what frame you're in, or what the units are. The docstring does, so that's OK, but I think it adds noise to the interface to make this a top-level attribute that's less clear than other attributes. I'm perfectly fine keeping it as a convenience method of course.

Reply:
Given that I'm mostly thinking about this in the context of convenience for various utilities, we could name it `_xyz`, which I think sould resolve the "clutter" issue while knocking down need for redundant code to craft `EarthLocation` into a 3-element array for various matrix operations.

Reply:
So it's currently available at `Telescope._location.xyz()` and I use it several places in the code. Does that work for you @kartographer?

Reply:
@bhazelton perfect!